tests/test_code_report.py

```diff
@@ -61,25 +76,32 @@ def test_without_code_report_command(runner_with_pylint):
     assert not pattern.findall(log)
 
 
-@pytest.mark.parametrize('args, expected_log', [
-    [["--python-version", python_version(), "--files", "source_file.py", "--result-file", "${CODE_REPORT_FILE}", '--rcfile'],
-     'rcfile: expected one argument'],
+@pytest.mark.parametrize('analyzer', ['pylint', 'mypy'])
+@pytest.mark.parametrize('common_arg_set, expected_log', [
     [["--python-version", python_version(), "--files", "source_file.py", "--result-file"],
      'result-file: expected one argument'],
     [["--python-version", python_version(), "--files", "--result-file", "${CODE_REPORT_FILE}"],
      "files: expected at least one argument"],
-
     [["--python-version", "--files", "source_file.py", "--result-file", "${CODE_REPORT_FILE}"],
      "python-version: expected one argument"],
     [["--python-version", python_version(), "--result-file", "${CODE_REPORT_FILE}"],
      "error: the following arguments are required: --files"],
 ])
-def test_pylint_analyzer_wrong_params(runner_with_pylint, args, expected_log):
-    source_file = runner_with_pylint.local.root_directory.join("source_file.py")
+def test_pylint_analyzer_wrong_common_params(runner_with_analyzers, analyzer, common_arg_set, expected_log):
+    test_pylint_analyzer_wrong_specific_params(runner_with_analyzers, analyzer, common_arg_set, expected_log)
```

Comment:
I couldn't find a way to do "AxB + C" with pytest fixtures, so went with a bit hacky way

Reply:
I am not sure pytest allows to call tests as regular functions.
As far as I know, we don't do that.

Reply:
it seems to work, alternatively I can move the code to separate function run_with_params and call that twice